pkg/distributor/distributor.go

```diff
@@ -772,7 +772,7 @@ func (d *Distributor) prePushHaDedupeMiddleware(next push.Func) push.Func {
 
 		removeReplica, err := d.checkSample(ctx, userID, cluster, replica)
 		if err != nil {
-			if errors.Is(err, replicasNotMatchError{}) {
+			if errors.Is(err, util.ReplicasNotMatchError{}) {
```

Comment:
This should be:
```suggestion
			if errors.As(err, &util.ReplicasNotMatchError{}) {
```

Then we don't need the `Is()` method.